packages/contracts-bedrock/src/L1/OPPrestateUpdater.sol

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Contracts
+import { OPContractsManager } from "src/L1/OPContractsManager.sol";
+
+// Interfaces
+import { ISuperchainConfig } from "interfaces/L1/ISuperchainConfig.sol";
+import { IProtocolVersions } from "interfaces/L1/IProtocolVersions.sol";
+import { IPermissionedDisputeGame } from "interfaces/dispute/IPermissionedDisputeGame.sol";
+import { IDisputeGameFactory } from "interfaces/dispute/IDisputeGameFactory.sol";
+import { IFaultDisputeGame } from "interfaces/dispute/IFaultDisputeGame.sol";
+import { IDelayedWETH } from "interfaces/dispute/IDelayedWETH.sol";
+import { IProxyAdmin } from "interfaces/universal/IProxyAdmin.sol";
+
+// Libraries
+import { Claim, GameTypes } from "src/dispute/lib/Types.sol";
+
+///  @title OPPrestateUpdater
+///  @notice A custom implementation of OPContractsManager that allows for modified deployment parameters
+contract OPPrestateUpdater is OPContractsManager {
+    /// @notice Thrown when a function from the parent (OPCM) is not implemented.
+    error NotImplemented();
+
+    /// @notice Thrown when the prestate of a permissioned disputed game is 0.
+    error PDGPrestateRequired();
+
+    /// @notice Thrown when the address off a fault dispute game is 0.
+    error FDGNotFound();
+
+    // @return Version string
+    /// @custom:semver 1.0.0
+    function version() public pure override returns (string memory) {
+        return "1.0.0";
+    }
+
+    // @notice Constructs the CustomOPContractsManager contract
+    // @param _superchainConfig Address of the SuperchainConfig contract
+    // @param _protocolVersions Address of the ProtocolVersions contract
+    // @param _l1ContractsRelease Version string for L1 contracts release
+    // @param _blueprints Addresses of Blueprint contracts
+    // @param _implementations Addresses of implementation contracts
+    // @param _upgradeController Address of the upgrade controller
+    constructor(
+        ISuperchainConfig _superchainConfig,
+        IProtocolVersions _protocolVersions,
+        IProxyAdmin _superchainProxyAdmin,
+        string memory _l1ContractsRelease,
+        Blueprints memory _blueprints,
+        Implementations memory _implementations,
+        address _upgradeController
+    )
+        OPContractsManager(
+            _superchainConfig,
+            _protocolVersions,
+            _superchainProxyAdmin,
+            _l1ContractsRelease,
+            _blueprints,
+            _implementations,
+            _upgradeController
+        )
+    { }
+
+    /// @notice Overrides the l1ContractsRelease function to return "none", as this OPCM
+    /// is not releasing new contracts.
+    function l1ContractsRelease() external pure override returns (string memory) {
+        return "none";
+    }
```

Comment:
This should probably return `op-contracts/v1.8.0`, since the way we change prestate is by deploying a new dispute game, and the dispute game is deployed via blueprints which are tied to a particular L1 contracts release?

Reply:
My concern with this is that as we redefine our release process, returning 1.8.0 here should indicate that the addresses in the `Implementation` struct correspond the 1.8.0 release, which they do not...

Reply:
> we want some way to know which version of the contracts this should be used with

Do you think we need this onchain, which was my main motivation for the suggestion. On second thought we probably not, since we agreed OPCM contracts will be maintained in an offchain registry